bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java

```diff
@@ -60,14 +88,18 @@ public Object[] getEnums() {
             // array element in JCR
             // todo: and compute based on it (hence using typeJcr below)
             // may expose internal representation of mutable object, hence cloning
-            return ComponentUtils.coerce(type, enums);
+            Map<Object, String> map = removeDuplicates();
+            String[] enumValue = map.keySet().toArray(new String[0]);
+            return ComponentUtils.coerce(type, enumValue);
         }
     }
 
     @Override
     public String[] getEnumNames() {
         if (enumNames != null) {
-            return Arrays.stream(enumNames)
+            Map<Object, String> map = removeDuplicates();
```

Comment:
Please add test cases for this use-case in junit. Also add a cypress test for runtime.

Reply:
Sure. Thanks